shared/utils/il2cpp-utils.cpp

```diff
@@ -42,19 +70,65 @@ namespace il2cpp_utils {
         return msg;
     }
 
-    bool ParameterMatch(const MethodInfo* method, Il2CppType** type_arr, int count) {
+    std::vector<const Il2CppType*> ClassVecToTypes(std::vector<const Il2CppClass*> seq) {
+        il2cpp_functions::Init();
+
+        std::vector<const Il2CppType*> types(seq.size());
+        std::transform(seq.begin(), seq.end(), types.begin(), il2cpp_functions::class_get_type_const);
+        return types;
+    }
+
+    bool ParameterMatch(const MethodInfo* method, const Il2CppType* const* type_arr, decltype(MethodInfo::parameters_count) count) {
         il2cpp_functions::Init();
         if (method->parameters_count != count) {
             return false;
         }
-        for (int i = 0; i < method->parameters_count; i++) {
+        for (decltype(method->parameters_count) i = 0; i < method->parameters_count; i++) {
```

Comment:
Why use decltype here when method->parameters_count is a relatively consistent typedef?
Seems a bit like artificial bloat